perf_test/blas/blas2/KokkosBlas2_gemv_perf_test.cpp

```diff
@@ -0,0 +1,204 @@
+  std::cerr << "\t[Optional] --layout :: matrix layout ('left' or 'right', default 'left')" << std::endl;
+  std::cerr << "\t[Optional] --m      :: number of rows to generate" << std::endl;
+  std::cerr << "\t[Optional] --n      :: number of cols to generate" << std::endl;
+}
+
+int parse_inputs (Params& params, int argc, char **argv){
+  for ( int i = 1 ; i < argc ; ++i ) {
+    if ( 0 == strcasecmp( argv[i] , "--help") || 0 == strcasecmp( argv[i] , "-h" )) {
+      print_options();
+      exit(0);  //note: this is before Kokkos::initialize
+    }
+    else if ( 0 == strcasecmp( argv[i] , "--threads" ) ) {
+      params.use_threads = atoi( argv[++i] );
+    }
+    else if ( 0 == strcasecmp( argv[i] , "--openmp" ) ) {
+      params.use_openmp = atoi( argv[++i] );
+    }
+    else if ( 0 == strcasecmp( argv[i] , "--cuda" ) ) {
+      params.use_cuda = atoi( argv[++i] ) + 1;
+    }
+    else if ( 0 == strcasecmp( argv[i] , "--layout" ) ) {
+      i++;
+      if(0 == strcasecmp( argv[i] , "left"))
+        params.layoutLeft = true;
+      else if(0 == strcasecmp( argv[i] , "right"))
+        params.layoutLeft = false;
+      else
+      {
+        std::cerr << "Invalid layout: must be 'left' or 'right'.\n";
+        exit(1);
+      }
+    }
+    else if( 0 == strcasecmp( argv[i], "--m" ))
+    {
+      params.m = atoi(argv[++i]);
+    }
+    else if( 0 == strcasecmp( argv[i], "--n" ))
+    {
+      params.n = atoi(argv[++i]);
+    }
+    else if ( 0 == strcasecmp( argv[i] , "--repeat" ) ) {
+      //if provided, C will be written to given file.
+      //has to have ".bin", or ".crs" extension.
+      params.repeat = atoi( argv[++i] );
+    }
+    else {
+      std::cerr << "Unrecognized command line argument #" << i << ": " << argv[i] << std::endl ;
+      print_options();
+      return 1;
+    }
+  }
+  return 0;
+}
+
+template<typename ExecSpace, typename Layout>
+void run(int m, int n, int repeat)
+{
+  using Scalar = double;
+  using MemSpace = typename ExecSpace::memory_space;
+  using Device = Kokkos::Device<ExecSpace, MemSpace>;
+  std::cout << "Running GEMV experiment (" << ExecSpace::name() << ")\n";
+  Kokkos::View<Scalar**, Layout, Device> A(Kokkos::ViewAllocateWithoutInitializing("A"), m, n);
+  Kokkos::View<Scalar*, Device> x(Kokkos::ViewAllocateWithoutInitializing("x"), n);
+  Kokkos::View<Scalar*, Device> y(Kokkos::ViewAllocateWithoutInitializing("y"), m);
+  Kokkos::Random_XorShift64_Pool<ExecSpace> pool(123);
+  Kokkos::fill_random(A, pool, 10.0);
+  Kokkos::fill_random(x, pool, 10.0);
+  //Do a warm-up run
+  KokkosBlas::gemv("N", 1.0, A, x, 0.0, y);
+  //Now, start timing
+  Kokkos::fence();
+  Kokkos::Timer timer;
+  for(int i = 0; i < repeat; i++)
+  {
+    KokkosBlas::gemv("N", 1.0, A, x, 0.0, y);
+    ExecSpace().fence();
+  }
+  double total = timer.seconds();
+  double avg = total / repeat;
+  size_t flopsPerRun = (size_t) m * n;
```

Comment:
This should 2 mn.

Reply:
I was counting a multiply-add as a single flop, but I see that the rest of KokkosKernels counts that as 2. So I'm happy to change it.

Reply:
And I'm not counting scales by alpha/beta since this test uses alpha=1, beta=0 and so an implementation could skip those.